frontend/src/api/k8s/__tests__/configMaps.spec.ts

```diff
@@ -0,0 +1,244 @@
+      createConfigMap(assembleConfigMap(projectName, configMapData, configMapName)),
+    ).rejects.toThrow('error1');
+    expect(k8sCreateResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sCreateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        plural: 'configmaps',
+      },
+      queryOptions: { queryParams: {} },
+      resource: configMapMock,
+    });
+  });
+});
+
+describe('replaceConfigMap', () => {
+  it('should successfully replace configMap', async () => {
+    k8sUpdateResourceMock.mockResolvedValue(configMapMock);
+
+    const result = await replaceConfigMap(
+      assembleConfigMap(projectName, configMapData, configMapName),
+    );
+    expect(k8sUpdateResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sUpdateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        plural: 'configmaps',
+      },
+      queryOptions: { queryParams: {} },
+      resource: configMapMock,
+    });
+    expect(result).toStrictEqual(configMapMock);
+  });
+
+  it('should handle errors and rethrow', async () => {
+    k8sUpdateResourceMock.mockRejectedValue(new Error('error1'));
+
+    await expect(
+      replaceConfigMap(assembleConfigMap(projectName, configMapData, configMapName)),
+    ).rejects.toThrow('error1');
+    expect(k8sUpdateResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sUpdateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        plural: 'configmaps',
+      },
+      queryOptions: { queryParams: {} },
+      resource: configMapMock,
+    });
+  });
+});
+
+describe('deleteConfigMap', () => {
+  it('should successfully delete configMap', async () => {
+    const mockK8sStatus: K8sStatus = {
+      apiVersion: 'v1',
+      kind: 'Status',
+      code: 200,
+      message: '200 OK',
+      reason: '',
+      status: 'Success',
+    };
+    k8sDeleteResourceMock.mockResolvedValue(mockK8sStatus);
+
+    const result = await deleteConfigMap(projectName, configMapName);
+    expect(result.status).toStrictEqual(mockK8sStatus.status);
+    expect(k8sDeleteResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sDeleteResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        plural: 'configmaps',
+      },
+      queryOptions: {
+        name: configMapName,
+        ns: projectName,
+        queryParams: {},
+      },
+    });
+  });
+
+  it('should return status failure when unsuccessful', async () => {
+    k8sDeleteResourceMock.mockResolvedValue(mock404Error({}));
+
+    const result = await deleteConfigMap(projectName, configMapName);
+    expect(result.status).toStrictEqual(mock404Error({}).status);
```

Comment:
You should create the mock once and reference it elsewhere in the same test function.
Also is there a reason the `result.status` is specifically being compared instead of the entire result?

```suggestion
    const mockStatus = mock404Error({});
    k8sDeleteResourceMock.mockResolvedValue(mockStatus);

    const result = await deleteConfigMap(projectName, configMapName);
    expect(result.status).toStrictEqual(mockStatus.status);
```